TooltipDecorator/useTooltip.js

```diff
@@ -0,0 +1,266 @@
+/* global MutationObserver ResizeObserver */
+
+import {forProp, forward, handle, not} from '@enact/core/handle';
+import {Job} from '@enact/core/util';
+import {useI18nContext} from '@enact/i18n/I18nDecorator';
+import {FloatingLayerBase} from '@enact/ui/FloatingLayer';
+import ri from '@enact/ui/resolution';
+
+import {Tooltip, defaultArrowAnchor, defaultDirection} from './Tooltip';
+import {adjustDirection, adjustAnchor, calcOverflow, getLabelOffset, getPosition} from './util';
+
+let currentTooltip; // needed to know whether or not we should stop a showing job when unmounting
+
+import useHandlers from '@enact/core/useHandlers';
+import {useCallback, useEffect, useLayoutEffect, useRef, useState} from 'react';
+
+function getDirectionAnchor (tooltipPosition, tooltipType) {
+	const position = tooltipPosition || (defaultDirection(tooltipType) + ' ' + defaultArrowAnchor(tooltipType));
+	const arr = position.split(' ');
+
+	if (arr.length === 2) {
+		return {
+			tooltipDirection: arr[0],
+			arrowAnchor: arr[1]
+		};
+	} else {
+		return {
+			tooltipDirection: position === 'below' ? 'below' : 'above',
+			arrowAnchor: 'right'
+		};
+	}
+}
+
+const removeTooltipProps = ({...props}) => {
+	delete props.rtl;
+	delete props.screenEdgeKeepout;
+	delete props.tooltipDelay;
+	delete props.tooltipMarquee;
+	delete props.tooltipPosition;
+	delete props.tooltipProps;
+	delete props.tooltipRelative;
+	delete props.tooltipText;
+	delete props.tooltipType;
+	delete props.tooltipUpdateDelay;
+	delete props.tooltipWidth;
+
+	return props;
+};
+
+// A hook to show Sandstone-styled tooltip components.
+function useTooltip (props = {}) {
+	const {
+		screenEdgeKeepout = (24 + 24), // Do NOT forget to check TooltipDecorator's default config value also.
+		tooltipDelay = 500, tooltipType = 'balloon', tooltipUpdateDelay = 400,
+		tooltipMarquee, tooltipPosition, tooltipProps, tooltipRelative, tooltipText, tooltipWidth
+	} = props;
+	const rtl = useI18nContext()?.rtl;
```

Comment:
`rtl` can be provided via prop from `TooltipDecorator` right?
Would you please consider that case too?

Reply:
Yes, `rtl` is passed as a part of TooltipDecorator.`rtl` is pased. And, no, we can't assume that as a standalone hook. That is why I use `useI18nContext` here.

Reply:
I updated to use `props.rtl` if a boolean value is given after discussion with Mikyung as it is a natural usage.